### PR TITLE
Add API methods to enable RUM trace correlation

### PR DESCRIPTION
**Pull request by felixbarny**

### CODE REVIEW

File: apm-agent-plugins/apm-api-plugin/src/main/java/co/elastic/apm/plugin/api/TransactionInstrumentation.java [hunk 1]
```diff
@@ -77,4 +78,23 @@ public static void setUser(@Advice.FieldValue(value = "span", typing = Assigner.
             transaction.setUser(id, email, username);
         }
     }
+
+    public static class MakeChildOfRumTransactionInstrumentation extends TransactionInstrumentation {
+        public MakeChildOfRumTransactionInstrumentation() {
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-api-plugin/src/main/java/co/elastic/apm/plugin/api/TransactionInstrumentation.java]: Stay consistent with naming based on the instrumented method name

File: apm-agent-plugins/apm-api-plugin/src/test/java/co/elastic/apm/api/ElasticApmApiInstrumentationTest.java [hunk 2]
```diff
@@ -162,4 +165,16 @@ void testScopes() {
         assertThat(ElasticApm.currentTransaction()).isSameAs(NoopTransaction.INSTANCE);
 
     }
+
+    @Test
+    void testMakeChildOfRumTransaction() {
```
Comment (eyalkoren) [inline apm-agent-plugins/apm-api-plugin/src/test/java/co/elastic/apm/api/ElasticApmApiInstrumentationTest.java]: use `ensureParentId` instead

File: apm-agent-plugins/apm-api-plugin/src/main/java/co/elastic/apm/plugin/api/TransactionInstrumentation.java (hunk 1 above)
Comment (felixbarny) [reply apm-agent-plugins/apm-api-plugin/src/main/java/co/elastic/apm/plugin/api/TransactionInstrumentation.java]: done

File: apm-agent-plugins/apm-api-plugin/src/test/java/co/elastic/apm/api/ElasticApmApiInstrumentationTest.java (hunk 2 above)
Comment (felixbarny) [reply apm-agent-plugins/apm-api-plugin/src/test/java/co/elastic/apm/api/ElasticApmApiInstrumentationTest.java]: done

File: apm-agent-api/src/main/java/co/elastic/apm/api/TransactionImpl.java [hunk 3]
```diff
@@ -40,4 +40,11 @@ public void setUser(String id, String email, String username) {
         // co.elastic.apm.plugin.api.TransactionInstrumentation$SetUserInstrumentation.setUser
     }
 
+    @Nonnull
+    @Override
+    public String ensureParentId() {
+        // co.elastic.apm.plugin.api.TransactionInstrumentation.MakeChildOfRumTransactionInstrumentation
```
Comment (eyalkoren) [inline apm-agent-api/src/main/java/co/elastic/apm/api/TransactionImpl.java]:
```suggestion
        // co.elastic.apm.plugin.api.TransactionInstrumentation.EnsureParentIdInstrumentation
```

**Pull request merged**